dashboard/test/lib/pd/survey_pipeline/daily_survey_joiner_test.rb

```diff
@@ -0,0 +1,95 @@
+require 'test_helper'
+require 'pd/survey_pipeline/daily_survey_joiner.rb'
+
+module Pd::SurveyPipeline
+  class DailySurveyJoinerTest < ActiveSupport::TestCase
+    include Pd::JotForm::Constants
+
+    self.use_transactional_test_case = true
+
+    setup_all do
+      form_id = 11_000_000_000_000
+      submission_id = 1
+
+      @question_content = {
+        'Q1' => {type: 'number', name: 'overallRating', text: 'Overall rating',
+          answer_type: ANSWER_TEXT},
+        'Q2' => {type: 'dropdown', name: 'selectOption', text: 'Select one of the options',
+          options: ['Option 1', 'Option 2', 'Option 3'],
+          option_map: {'Option 1' => 1, 'Option 2' => 2, 'Option 3' => 3},
+          answer_type: ANSWER_SINGLE_SELECT},
+        'Q3' => {type: 'textarea', name: 'describe', text: 'Describe something',
+          answer_type: ANSWER_TEXT},
+        'Q4' => {type: 'matrix', name: 'multiQuestions', text: 'Answer all sub questions',
+          sub_questions: ['Sub question 1', 'Sub question 2', 'Sub question 3'],
+          options: ['Option 1', 'Option 2', 'Option 3'],
+          option_map: {'Option 1' => 1, 'Option 2' => 2, 'Option 3' => 3},
+          answer_type: ANSWER_MULTI_SELECT}
+      }
+
+      @questions = {
+        form_id => @question_content
+      }
+
+      @submissions_context = {
+        workshop_id: nil,
+        user_id: nil,
+        day: nil,
+        session_id: nil,
+        facilitator_id: nil
+      }
+
+      @submissions = {
+        form_id => {
+          submission_id => @submissions_context.merge(
+            answers: {
+              'Q1' => '5.0',
+              'Q2' => 'Option 2',
+              'Q3' => 'Like it',
+              'Q4' => [
+                {text: 'Sub question 1', answer: 'Option 1'},
+                {text: 'Sub question 2', answer: 'Option 2'},
+                {text: 'Sub question 3', answer: 'Option 3'}
+              ]
+            }
+          )
+        }
+      }
+
+      @submissions_context.merge!(form_id: form_id, submission_id: submission_id)
+    end
+
+    test 'can join submission and question' do
+      # Expected flatten results for non-matrix questions
```

Comment:
You've got a large `setup_all` for one test case here!

I'd like to see the matrix question behavior in a separate test case from other question types.

Reply:
The expectation is more tests will be added later. I can move the setup into the test and later split it out if it's going to be shared among tests.

Reply:
Split into 2 tests, 1 for non-matrix and 1 for matrix question.